sdk/core/Azure.Core/tests/TestFramework/PlaybackTransport.cs

```diff
@@ -14,21 +15,62 @@ public class PlaybackTransport : MockTransport
 
         private readonly RecordMatcher _matcher;
 
-        public PlaybackTransport(RecordSession session, RecordMatcher matcher)
+        private readonly Random _random;
+
+        public PlaybackTransport(RecordSession session, RecordMatcher matcher, Random random)
         {
             _session = session;
             _matcher = matcher;
+            _random = random;
         }
 
         public override void Process(HttpPipelineMessage message)
         {
+            // Some tests will check if the Request Content is being read (to
+            // verify their Progress handling) so we'll just copy it to a
+            // MemoryStream
+            if (message.Request.Content != null &&
+                message.Request.Content.TryComputeLength(out long length) &&
+                length > 0)
+            {
+                using (MemoryStream stream = new MemoryStream((int)length))
+                {
+                    message.Request.Content.WriteTo(stream, message.CancellationToken);
+                }
+            }
+
             message.Response = GetResponse(_session.Lookup(message.Request, _matcher));
         }
 
-        public override Task ProcessAsync(HttpPipelineMessage message)
+        public override async Task ProcessAsync(HttpPipelineMessage message)
         {
+            // Some tests will check if the Request Content is being read (to
+            // verify their Progress handling) so we'll just copy it to a
+            // MemoryStream asynchronously
+            if (message.Request.Content != null &&
+                message.Request.Content.TryComputeLength(out long length) &&
+                length > 0)
+            {
+                using (MemoryStream stream = new MemoryStream((int)length))
+                {
+                    await message.Request.Content.WriteToAsync(stream, message.CancellationToken).ConfigureAwait(false);
+                }
+            }
+
             message.Response = GetResponse(_session.Lookup(message.Request, _matcher));
-            return Task.CompletedTask;
+        }
+
+        public override Request CreateRequest()
+        {
+            lock (_random)
+            {
+                // Force a call to random.NewGuid so we keep the random seed
+                // unified between record/playback
+                _random.NewGuid();
+
+                // TODO: Pavel will think about ways to unify this
```

Comment:
should we add an issue here instead? so we don't forget to do it?

Reply:
Yes - good catch.  Issue #6615 has been added.